testingfacade/APIs/schemas/question-schema.json

```diff
@@ -17,34 +18,35 @@
     "properties": {
         "test_type": {
             "type": "string",
-            "description": "The type of question being sent indicating the type of answer that needs to be returned.",
-            "enum": ["single_choice", "multi_choice", "action"]
+            "description": "The type of question being sent indicating the type of answer that needs to be returned",
+            "enum": [
+                "single_choice", 
+                "multi_choice", 
+                "action"
+            ]
         },
         "question_id": {
             "description": "A unique string for each individual question sent. Where a test contains multiple questions they will share a name attribute but have a unique id.",
             "anyOf": [
                 {
                     "description": "Identifier for messages sent at the beginning and end of every set of tests",
+                    "type": "string",
                     "enum": [
                         "pre_tests_message",
                         "post_tests_message"
                     ]
                 },
                 {
-                    "description": "id for questions within a single part test",
-                    "type": "string",
-                    "pattern": "test_[0-9]+"
-                },
-                {
-                    "description": "id for questions within a multi part test",
+                    "description": "Identifier for questions within a test",
                     "type": "string",
-                    "pattern": "test_[0-9]+_[0-9]+"
+                    "pattern": "test_[0-9a-z_]+"
```

Comment:
Hi, @hrbriggs. Please would you add `^`...`$` round the patterns for the test, question and answer IDs, like the resource ID one. This ensures it's not just an internal match, like `meow-test_1-purr`.

Reply:
Done